### DISCUSSION TITLE
Find channel across all public teams

### DISCUSSION THREAD

**Discussion by bbodenmiller** [mattermost/community#118] [Practical Ideas]
Let's say I'm on a server with dozens of teams and I'm looking for a JavaScript channel. Would be great if I could easily search all channel metadata regardless of team (and whether I've already joined it or not) so for example I could find all channels about JavaScript.

Motivation: the issue I'm seeing is people creating redundant channels because they didn't know one already existed that they could access for their given topic.

**Comment by esethna** [comment]
@bbodenmiller can you expand on what you mean by "search all channel metadata". Currently the channel switcher will find public channels by name that your not a member of yet (just not across teams yet)

**Reply by bbodenmiller** [reply]
@esethna essentially I mean search channel names across all teams I am a member of or have permissions to join. I could see search channel header or description as helpful too.

**Reply by esethna** [reply]
Got it, this idea was shared in the context of the channel switcher but the browse channel modal could likely benefit from this too
